web-gateway/src/main/java/controllers/UserController.java

```diff
@@ -61,4 +61,29 @@ public Result currentUser(String userId) {
         session("user", userId);
         return ok("User switched");
     }
+    public CompletionStage<Result> loginUser() {
+        Http.Context ctx = ctx();
+        return withUser(ctx, userId ->
+                loadNav(userId).thenCompose(nav -> {
+                    Form<LoginForm> form = formFactory.form(LoginForm.class).bindFromRequest(ctx.request());
+                    if (form.hasErrors()) {
+                        return CompletableFuture.completedFuture(ok(views.html.login.render(showInlineInstruction, form, nav)));
+                    }
+
+                    return userService.login().invoke(new UserRegistration(nav.getUser().get().getName(), form.get().getEmail(), form.get().getPassword())).thenApply(id -> {
```

Comment:
@lakhina I don't think the `login` operation should reuse the `UserRegistration`. It contains a `name` field which is not available: because `nav.getUser().get().getName()` can't be resolved since there's no logged in user.